### ISSUE TITLE
Freeze an app with a different method when holding the freeze button

### ISSUE THREAD

**Issue by ghost** [MuntashirAkon/AppManager#1449]
### Please check before submitting an issue

- [X] I am using the latest version of App Manager
- [X] I have searched the issues and haven't found anything relevant
- [X] I have read the docs

### Describe a description of the new feature

You should add buttons to the menu that shows when you hold the Freeze button to freeze an app with a different method.

### Describe the solution you'd like

Add new buttons to the Freeze/unfreeze menu to freeze an app with a different method.

**Comment by shuvashish76** [comment]
So your feature request is :
- Add the freezing method "Suspend, Disable, Hide" options as chips _(just like how it's used for "sort" in Main page)_ for individual apps when you long click ["Freeze" option from "Horizontal Action Panel" of "App Info" page](https://muntashirakon.github.io/AppManager/en/#subsubsec:horizontal-action-panel), without each time changing default option from "Settings > Rules > Default freezing method"
- And under that show all the checkboxs as we currently show.
- Remember the different Freezing methods used for all the apps.

---

This feature request is similar to how we've freedom of individual configuration for "IFW+Dis, IFW, Dis" component blocking methods, overriding Default options.

Note: Similar to #1201, we already show the correct "[Freezing status Tag](https://muntashirakon.github.io/AppManager/en/#subsubsec:tags)" for individual apps.

**Comment by MuntashirAkon** [comment]
I've been thinking about this feature myself for quite some time. The problem is the long-click option has already been used for creating shortcuts. I cannot think of a good way to offer both features at the same time.

**Comment by MuntashirAkon** [comment]
> 3\. Remember the different Freezing methods used for all the apps.

This could be problematic though. Because unless there's a way to report to the user about the freezing method used for the app, it might be difficult for them to figure out (in advance) which freezing method is going to be used for this particular app (since we don't expect them to remember the last used freezing method for an app). But, there are also some benefits, especially for batch operations and profiles.

**Comment by shuvashish76** [comment]
> it might be difficult for them to figure out (in advance) **which freezing method is going to be used** for this particular app (since we don't expect them to remember the last used freezing method for an app).

Yes I agree. I thought "_Hit-and-Trial_" method shouldn't be a problem as you don't lose any data with any of the freezing method.
1. **(a)** Just click Freeze to check what tag appears on Horizontal action panel. If it's different than your intended method, simply unfreeze and **(b)** Long click to view/change which chip used for the freezing method.
2. If you want to know what freezing method is already set before performing the action, then follow **1.(b)**

**Comment by shuvashish76** [comment]
>  there are also some benefits, especially for batch operations and profiles.

- For "Freeze" batch operations, show 4 options in the dialog (Suspend, Disable, Hide and **User preferred** or something similar name...which would follow your default freezing method set from the app + user preferred method set for individual apps)
- Give similar 4 options for Profiles > open any profile > Configurations > Freeze

**Comment by shuvashish76** [comment]
> User preferred or something similar name...which would follow your default freezing method set from the AM for all apps + (except) user preferred method set for specific apps)

> If you choose none of the options and click Freeze then follow only default freezing method set from the AM for all apps.

Limitation of my Freeze" batch operations proposal : Suppose you want to use Suspend (which is not your default freezing method) + User preferred method. Then it's not possible with current proposal.

Alternative idea:
![image](https://github.com/user-attachments/assets/c63b0888-9c0f-4507-b811-3d9203a0f159)

This way you'll have the freedom to do
- Only Suspend/Disable/Hide
- Suspend/Disable/Hide + User preferred
- If you want to go AM default method then unselect all options and click Freeze.

**Comment by MuntashirAkon** [comment]
> i. **(a)** Just click Freeze to check what tag appears on Horizontal action panel. If it's different than your intended method, simply unfreeze and **(b)** Long click to view/change which chip used for the freezing method.

Technically, if an app is frozen with any methods, it needs to be unfrozen again before applying any other method. Because App Manager will freeze again with the new method without checking whether it was frozen before. This is because App Manager does not know if it was previously frozen with App Manager (the system does not supply a timestamp for these operations).

**Comment by MuntashirAkon** [comment]
> The problem is the long-click option has already been used for creating shortcuts. I cannot think of a good way to offer both features at the same time.

With #1426, we can actually do this more efficiently. Instead of a "Are you sure?" prompt, we can display a list of options (suspend, disable, hide) to the user.

```
 ___________________________
| Choose what to do:        |
|                           |
| - [ ] Suspend             |
| - [ ] Advanced suspend    |
| - [x] Disable             |
| - [ ] Hide                |
|                           |
| [ ] Remember for this app |
|                           |
|          Cancel    Freeze |
'''''''''''''''''''''''''''''
```

The remember option is checked automatically if the user has already set up a manual option.

**Comment by MuntashirAkon** [comment]
Implemented via f6b70382112c055b3691ca19dd0bd99456695820
